src/sema/expression.rs

```diff
@@ -6187,14 +6204,8 @@ fn resolve_using(
             let mut matches = true;
             let mut cast_args = Vec::new();
 
-            match cast(
-                &self_expr.loc(),
-                self_expr.clone(),
-                &libfunc.params[0].ty,
-                true,
-                ns,
-                &mut errors,
-            ) {
+            let param_ty = ns.functions[function_no].params[0].ty.clone();
+            match self_expr.cast(&self_expr.loc(), &param_ty, true, ns) {
                 Ok(e) => cast_args.push(e),
                 Err(()) => continue,
             }
```

Comment:
Here we try to convert the self expression to the first argument of the library function, and if this fails, we should try the next one without failing. Now any mismatch becomes a compiler error, which is broken. I've written an example here:

This example fails to compile on your branch but works fine on the main branch:
```
library lib1 {
	function fmuladd(int a, int b, int c) internal pure returns (int) {
		return a * b + c;
	}

	function fmuladd(int a, int b, bool c) internal pure returns (int) {
		return a * b;
	}

}

library lib2 {
	function fmuladd(int a, int b, bool c) internal pure returns (int) {
		return a * b;
	}

	function fmuladd(uint a, uint b, uint c) internal pure returns (uint) {
		return a * b + c;
	}
}

contract bar {
	using lib1 for *;
	using lib2 for *;

	function f1(int a, int b) public pure returns (int) {
		return a.fmuladd(b, 2);	
	}

	function f2(uint a, uint b) public pure returns (uint) {
		return a.fmuladd(b, 2);	
	}

}
```

Reply:
Please add this example to test cases

Reply:
@LucasSte Please add this example to test cases

Reply:
Have a look at `sema/tests/mod.rs`